src/EnergyPlus/FaultsManager.cc

```diff
@@ -218,355 +277,747 @@ namespace FaultsManager {
+				// check availability schedule
+				if ( lAlphaFieldBlanks( 2 ) ) {
+					FaultsEconomizer( j ).AvaiSchedPtr = -1; // returns schedule value of 1
+				} else {
+					FaultsEconomizer( j ).AvaiSchedPtr = GetScheduleIndex( cAlphaArgs( 2 ) );
+					if ( FaultsEconomizer( j ).AvaiSchedPtr == 0 ) {
+						ShowSevereError( cFaultCurrentObject + " = \"" + cAlphaArgs( 1 ) + "\" invalid " + cAlphaFieldNames( 2 ) + " = \"" + cAlphaArgs( 2 ) + "\" not found." );
+						ErrorsFound = true;
 					}
+				}
 
-					Faults( j ).ControllerName = cAlphaArgs( 5 );
-					// check controller name
-					if ( lAlphaFieldBlanks( 5 ) ) {
-						ShowSevereError( cFault1 + " = \"" + cAlphaArgs( 1 ) + "\" invalid " + cAlphaFieldNames( 5 ) + " = \"" + cAlphaArgs( 5 ) + "\" blank." );
+				FaultsEconomizer( j ).SeveritySchedule = cAlphaArgs( 3 );
+				// check severity schedule
+				if ( lAlphaFieldBlanks( 3 ) ) {
+					FaultsEconomizer( j ).SeveritySchedPtr = -1; // returns schedule value of 1
+				} else {
+					FaultsEconomizer( j ).SeveritySchedPtr = GetScheduleIndex( cAlphaArgs( 3 ) );
+					if ( FaultsEconomizer( j ).SeveritySchedPtr == 0 ) {
+						ShowSevereError( cFaultCurrentObject + " = \"" + cAlphaArgs( 1 ) + "\" invalid " + cAlphaFieldNames( 3 ) + " = \"" + cAlphaArgs( 3 ) + "\" not found." );
 						ErrorsFound = true;
 					}
+				}
 
-					// offset - degree of fault
-					Faults( j ).Offset = rNumericArgs( 1 );
+				FaultsEconomizer( j ).ControllerType = cAlphaArgs( 4 );
+				// check controller type
+				if ( lAlphaFieldBlanks( 4 ) ) {
+					ShowSevereError( cFaultCurrentObject + " = \"" + cAlphaArgs( 1 ) + "\" invalid " + cAlphaFieldNames( 4 ) + " = \"" + cAlphaArgs( 4 ) + "\" blank." );
+					ErrorsFound = true;
+				} else {
+					{ auto const SELECT_CASE_var( MakeUPPERCase( cAlphaArgs( 4 ) ) );
+					if ( SELECT_CASE_var == "CONTROLLER:OUTDOORAIR" ) {
+						FaultsEconomizer( j ).ControllerTypeEnum = iController_AirEconomizer;
+
+						//CASE ...
+
+					} else {
+					}}
+				}
+
+				FaultsEconomizer( j ).ControllerName = cAlphaArgs( 5 );
+				// check controller name
+				if ( lAlphaFieldBlanks( 5 ) ) {
+					ShowSevereError( cFaultCurrentObject + " = \"" + cAlphaArgs( 1 ) + "\" invalid " + cAlphaFieldNames( 5 ) + " = \"" + cAlphaArgs( 5 ) + "\" blank." );
+					ErrorsFound = true;
 				}
+
+				// offset - degree of fault
+				FaultsEconomizer( j ).Offset = rNumericArgs( 1 );
 			}
 		}
-
-		RunMeOnceFlag = true;
+		
+		RunFaultMgrOnceFlag = true;
 
 		if ( ErrorsFound ) {
 			ShowFatalError( "Errors getting FaultModel input data.  Preceding condition(s) cause termination." );
 		}
 
 	}
 
-	bool
-	CheckFaultyAirFilterFanCurve(
-		std::string const & FanName, // name of the fan
-		int const FanCurvePtr      // pointer of the fan curve
+	Real64
+	FaultProperties::CalFaultOffsetAct()
+	{
+
+		// SUBROUTINE INFORMATION:
+		//       AUTHOR         Rongpeng Zhang
+		//       DATE WRITTEN   Jun. 2016
+		//       MODIFIED       na
```

Comment:
Minor detail - these na comments and other unused comment blocks below can be omitted.  The old comment template does not apply to new code.